cupy/testing/helper.py

```diff
@@ -101,8 +101,13 @@ def test_func(self, *args, **kw):
             # values are negative.
             if _contains_signed_and_unsigned(kw):
                 inds = _make_positive_indices(self, impl, args, kw)
-                cupy_result = cupy.asnumpy(cupy_result)[inds]
-                numpy_result = cupy.asnumpy(numpy_result)[inds]
+                try:
```

Comment:
Sorry, I confused.
I fixed code again.
https://github.com/pfnet/chainer/pull/1784/files#diff-409f5a9f6ca41e85a0ac7eaf1af60ba3R106
Your current code has bug.
If input is scalar and empty index(=`(array([], dtype=int64),)` ), current code is not correct working.
```python
a = numpy.zeros(())
numpy.nonzero(a >= 0) # (array([0]),)
a[numpy.nonzero(a >= 0)] # IndexError
numpy.nonzero(a >= 1) # (array([], dtype=int64),)
a[numpy.nonzero(a >= 1)] # IndexError
```